packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/deck/ViewDeckDescriptionConverter.java

```diff
@@ -108,8 +113,32 @@ public IRepresentationDescription convert(RepresentationDescription viewRepresen
             return new DeckStyleProvider(effectiveStyle).apply(variableManager);
         };
 
+        Function<VariableManager, List<String>> iconURLsProvider = variableManager -> this.getIconURLs(viewDeckDescription.getIconExpression(), variableManager, interpreter);
+
         return new org.eclipse.sirius.components.deck.description.DeckDescription(id, label, idProvider, labelProvider, this::getTargetObjectId, canCreatePredicate, laneDescriptions, dropLaneProvider,
-                styleProvider);
+                styleProvider, iconURLsProvider);
+    }
+
+    private List<String> getIconURLs(String iconExpression, VariableManager variableManager, AQLInterpreter interpreter) {
+        List<String> result = List.of();
+
+        if (iconExpression.isBlank()) {
+            result = this.representationImageProviders.stream()
+                    .map(representationImageProvider -> representationImageProvider.getImageURL(Deck.KIND))
+                    .flatMap(Optional::stream)
+                    .toList();
+        } else {
+            Object evaluated = this.evaluateObject(interpreter, variableManager, iconExpression);
```

Comment:
Just ask for the result as a list, no need to cast it manually afterward

Reply:
We thought that AQL service would mostly return a single path as a string.